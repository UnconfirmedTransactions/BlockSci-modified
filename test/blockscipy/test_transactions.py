import pytest
from util import correct_timestamp


def test_p2pkh_transaction_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-0-tx"])
    transaction_regression(tx, regtest)


def test_p2sh_transaction_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-1-tx"])
    transaction_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_transaction_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-2-tx"])
    transaction_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_transaction_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    transaction_regression(tx, regtest)


def transaction_regression(tx, regtest):
    print(tx.base_size, file=regtest)
    print(tx.block_height, file=regtest)
    print(correct_timestamp(tx.block_time), file=regtest)
    print(tx.fee, file=regtest)
    print(tx.fee_per_byte(), file=regtest)
    print(tx.fee_per_kbyte(), file=regtest)
    print(tx.hash, file=regtest)
    print(tx.index, file=regtest)
    print(tx.is_coinbase, file=regtest)
    print(tx.locktime, file=regtest)
    print(tx.observed_in_mempool, file=regtest)
    print(tx.op_return, file=regtest)
    print(tx.output_count, file=regtest)
    print(tx.output_value, file=regtest)
    print(tx.size_bytes, file=regtest)
    print(tx.time_seen, file=regtest)
    print(tx.total_size, file=regtest)
    print(tx.version, file=regtest)
    print(tx.virtual_size, file=regtest)
    print(tx.weight, file=regtest)
