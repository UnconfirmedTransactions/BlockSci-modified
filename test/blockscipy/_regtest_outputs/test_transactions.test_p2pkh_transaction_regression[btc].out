225
111
2018-09-01 00:20:00+00:00
10000
44
44444
9a27d8c02ed0ae0f62072963bae34ce845e7ba6b971d9bc76f58e14bec91c3e3
112
False
0
False
None
2
4999990000
225
None
225
1
225
900
